simple_salesforce/api.py

```diff
@@ -442,11 +443,15 @@ def __init__(
         * sf_instance -- the domain of the instance of Salesforce to use
         * sf_version -- the version of the Salesforce API to use
         * proxies -- the optional map of scheme to proxy server
+        * session -- Custom requests session, created in calling code. This
+                     enables the use of requests Session features not otherwise
+                     exposed by simple_salesforce.
         """
         self.session_id = session_id
         self.name = object_name
-        self.request = requests.Session()
-        self.request.proxies = proxies
+        self.request = session or requests.Session()
+        if proxies is not None:  # don't wipe out original proxies with None
+            self.request.proxies = proxies
```

Comment:
I think that if a session is passed in explicitly, `proxies` should be ignored. The side effect on the session instance owned by an upstream object concerns me a little.